apps/research-service/src/validations/validators/spikes.validator.ts

```diff
@@ -0,0 +1,165 @@
+import { Injectable } from "@nestjs/common";
+import { QueryTypes } from "sequelize";
+import { PolygonGeometry } from "@terramatch-microservices/database/entities";
+
+interface GeoJSONPolygon {
+  type: "Polygon";
+  coordinates: number[][][];
+}
+
+interface GeoJSONMultiPolygon {
+  type: "MultiPolygon";
+  coordinates: number[][][][];
+}
+
+type GeoJSONGeometry = GeoJSONPolygon | GeoJSONMultiPolygon;
+
+interface SpikeDetectionResult {
+  valid: boolean;
+  extraInfo: {
+    spikes: number[][];
+    spikeCount: number;
+  } | null;
+}
+
+@Injectable()
+export class SpikesValidator {
+  async validatePolygon(polygonUuid: string): Promise<SpikeDetectionResult> {
```

Comment:
I see that both of the validators in this PR have the same public interface (`validationPolygon` and `validatePolygons`) that seems likely to hold true for future validators are well - this should be extracted into a type or interface to make the typing on the proposed mapping in `validation.service.ts` more type safe.